amgut/lib/data_access/env_management.py

```diff
@@ -0,0 +1,387 @@
+from os.path import abspath, basename, dirname, join, split, splitext
+from glob import glob
+from functools import partial
+import subprocess
+
+from click import echo
+from psycopg2 import (connect, OperationalError, ProgrammingError)
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+from natsort import natsorted
+
+from amgut.lib.config_manager import AMGUT_CONFIG
+from amgut.lib.data_access.sql_connection import SQLConnectionHandler
+
+get_db_file = partial(join, join(dirname(dirname(abspath(__file__))), '..', 'db'))
+LAYOUT_FP = get_db_file('ag_unpatched.sql')
+INITIALIZE_FP = get_db_file('initialize.sql')
+POPULATE_FP = get_db_file('populate_test.sql')
+PATCHES_DIR = get_db_file('patches')
+
+
+def _check_db_exists(db, cursor):
+    r"""Check if the database db exists on the postgres server
+
+    Parameters
+    ----------
+    db : str
+        The database name
+    cursor : psycopg2.cursor
+        The cursor connected to the server
+    """
+    cursor.execute('SELECT datname FROM pg_database')
+    # It's a list of tuple, so just create the tuple to check if exists
+    return (db,) in cursor.fetchall()
+
+
+def create_database(force=False):
+    # Connect to the postgres server
+    try:
+        conn = connect(dbname='postgres',
+                       user=AMGUT_CONFIG.user, password=AMGUT_CONFIG.password,
+                       host=AMGUT_CONFIG.host, port=AMGUT_CONFIG.port)
+    except OperationalError as e:
+        raise OperationalError("Cannot connect to the server, error is %s" %
+                               str(e))
+
+    # Set the isolation level to AUTOCOMMIT so we can execute a create database
+    # sql query
+    conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+
+    # Get the cursor
+    cur = conn.cursor()
+
+    cur.execute('SET SEARCH_PATH TO ag, barcodes, public')
+
+    db_exists = _check_db_exists(AMGUT_CONFIG.database, cur)
+
+    # Check that the database does not already exist
+    if db_exists and force:
```

Comment:
I find this construction just slightly confusing, and would find a nested `if` clearer:
```python
if db_exists:
    if force:
        return
    raise ValueError("Database '{}' already present on the system".format(AMGUT_CONFIG.database))
```